src/tests/ftest/rebuild/continues_after_stop.py

```diff
@@ -78,48 +81,42 @@ def test_continuous_after_stop(self):
         ior_start_time = journalctl_time()
         thread.start()
 
-        # 3. After a few seconds, stop one of the ranks (rank 3).
         self.log_step("After a few seconds, stop one of the ranks (rank 3).")
         # Wait for IOR to start and write some data. Otherwise rebuild will not occur.
         time.sleep(5)
         self.server_managers[0].stop_ranks(ranks=[3], daos_log=self.log)
 
-        # 4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with daos_server
-        # identifier.
         msg = ("4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with "
                "daos_server identifier.")
         self.log_step(msg)
-        scanning_found = False
-        for count in range(120):
-            self.log.info("Look for 'Rebuild [scanning]'. Count = %d", count)
+
+        def _search_scanning():
+            """Search 'Rebuild [scanning]' from journalctl output using wait_for_result().
+            """
+            self.log.info("Search 'Rebuild [scanning]'. Count = %d", self.search_count)
+            self.search_count += 1
```

Comment:
FYI it's possible to do this without using `self`, so the variable is contained. This defines a static variable scoped to the helper method
```
def _search_scanning():
    self.log.info("Search 'Rebuild [scanning]'. Count = %d", _search_scanning.count)
    _search_scanning.count += 1
    ...
_search_scanning.count = 0
```

Also, I wonder if the count here is useful? `wait_for_result` does print each time it retries, but it doesn't show a count. I'll let you decide whether you need it :)
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14100/8/artifact/Functional%20Hardware%20Medium/rebuild/continues_after_stop.py/repeat001/job.log
```
2024-04-13 14:34:36,807 test             L0485 INFO | ==> Step 7: 4. Look for the start of the rebuild (Rebuild [scanning]) in journalctl with daos_server identifier. [elapsed since last step: 7.80s]
2024-04-13 14:34:36,807 general_utils    L1437 DEBUG| wait_for_result: Waiting for a result from _search_scanning() with a 120 second timeout
2024-04-13 14:34:36,807 continues_after_ L0096 INFO | Search 'Rebuild [scanning]'. Count = 0
2024-04-13 14:34:36,807 general_utils    L0554 INFO |   Obtaining journalctl data on wolf-[134-135]
2024-04-13 14:34:37,167 general_utils    L1443 DEBUG| wait_for_result: Result from _search_scanning(): False (timed out: False)
2024-04-13 14:34:38,168 continues_after_ L0096 INFO | Search 'Rebuild [scanning]'. Count = 1
2024-04-13 14:34:38,168 general_utils    L0554 INFO |   Obtaining journalctl data on wolf-[134-135]
2024-04-13 14:34:38,468 general_utils    L1443 DEBUG| wait_for_result: Result from _search_scanning(): False (timed out: False)
2024-04-13 14:34:39,469 continues_after_ L0096 INFO | Search 'Rebuild [scanning]'. Count = 2
2024-04-13 14:34:39,469 general_utils    L0554 INFO |   Obtaining journalctl data on wolf-[134-135]
...
```

Reply:
Thanks for the code. The count was useful while debugging to quickly find the number of iterations used, but now I'm done writing the code, so I guess we don't need extra code to print it anymore, so I removed it.